[x264] build failure

**Host Environment**
- OS: Windows
- Compiler: Visual Studio 2019

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install ffmpeg[avcodec,avformat,swresample,swscale,x264]:default`

**Failure logs** 
```
CMake Error at ports/x264/portfile.cmake:74 (file):
  file RENAME failed to rename

    C:/vcpkg/packages/x264_default/debug/lib/libx264.dll.lib

  to

    C:/vcpkg/packages/x264_default/debug/lib/libx264.lib

  because: The system cannot find the path specified.



Call Stack (most recent call first):
  scripts/ports.cmake:140 (include)


Error: Building package x264:default failed with: BUILD_FAILED
```

**Additional context**
This problem occurs when using non default triplet without building debug libraries, that directory just doesn't exist.

--- a/ports/x264/portfile.cmake
+++ b/ports/x264/portfile.cmake
@@ -71,7 +71,10 @@ endif()
 
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic" AND VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
     file(RENAME ${CURRENT_PACKAGES_DIR}/lib/libx264.dll.lib ${CURRENT_PACKAGES_DIR}/lib/libx264.lib)
-    file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/libx264.dll.lib ${CURRENT_PACKAGES_DIR}/debug/lib/libx264.lib)
+
+    if (NOT VCPKG_BUILD_TYPE)
+        file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/libx264.dll.lib ${CURRENT_PACKAGES_DIR}/debug/lib/libx264.lib)
+    endif()
 elseif(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
     # force U_STATIC_IMPLEMENTATION macro
     file(READ ${CURRENT_PACKAGES_DIR}/include/x264.h HEADER_CONTENTS)

--- a/ports/x264/vcpkg.json
+++ b/ports/x264/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "x264",
   "version-string": "164-5db6aa6cab1b146",
+  "port-version": 1,
   "description": "x264 is a free software library and application for encoding video streams into the H.264/MPEG-4 AVC compression format",
   "homepage": "https://github.com/mirror/x264",
   "supports": "!arm",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -7198,7 +7198,7 @@
     },
     "x264": {
       "baseline": "164-5db6aa6cab1b146",
-      "port-version": 0
+      "port-version": 1
     },
     "x265": {
       "baseline": "3.4",

--- a/versions/x-/x264.json
+++ b/versions/x-/x264.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "8d6c7ba5815a0683a2915df5f95de5d06e938781",
+      "version-string": "164-5db6aa6cab1b146",
+      "port-version": 1
+    },
+    {
       "git-tree": "2280334f1235046e20f80586b7d83893f52b23b7",
       "version-string": "164-5db6aa6cab1b146",
       "port-version": 0

